src/main/java/io/odpf/firehose/filter/JsonFilter.java

```diff
@@ -0,0 +1,176 @@
+        objectMapper = new ObjectMapper();
+        JsonSchemaFactory schemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
+
+        jsonPrinter = JsonFormat.printer().preservingProtoFieldNames();
+        messageType = consumerConfig.getFilterMessageType();
+        filterDataSourceType = consumerConfig.getFilterJsonDataSource();
+        protoSchemaClass = consumerConfig.getFilterJsonSchemaProtoClass();
+
+        filterJsonSchema = consumerConfig.getFilterJsonSchema();
+        this.instrumentation = instrumentation;
+        logConfigs();
+
+        if (filterDataSourceType != NONE) {
+            try {
+                schema = schemaFactory.getSchema(filterJsonSchema);
+            } catch (Exception e) {
+                instrumentation.logError("Failed to parse JSON Schema " + e.getMessage());
+            }
+        }
+
+        if (messageType == PROTOBUF) {
+            try {
+                protoParser = MethodUtils.getAccessibleMethod(Class.forName(protoSchemaClass), "parseFrom", byte[].class);
+            } catch (ClassNotFoundException e) {
+                instrumentation.logError("Failed to load Proto schema class " + e.getMessage());
+            }
+        }
+    }
+
+    private void logConfigs() {
+        instrumentation.logInfo("\n\tFilter type: {}", filterDataSourceType);
+        if (filterDataSourceType != NONE) {
+            instrumentation.logInfo("\n\tFilter JSON Schema: {}", filterJsonSchema);
+            instrumentation.logInfo("\n\tFilter message type: {}", messageType);
+
+            if (messageType == PROTOBUF) {
+                instrumentation.logInfo("\n\tMessage Proto class: {}", protoSchemaClass);
+            }
+        } else {
+            instrumentation.logInfo("No filter is selected");
+        }
+    }
+
+    /**
+     * method to filter the EsbMessages.
+     *
+     * @param messages the json/protobuf records in binary format that are wrapped in {@link Message}
+     * @return the list of filtered Messages
+     * @throws FilterException the filter exception
+     */
+    @Override
+    public List<Message> filter(List<Message> messages) throws FilterException {
+
+        if (filterDataSourceType == NONE) {
+            return messages;
+        }
+        validateConfigs();
+
+        List<Message> filteredMessages = new ArrayList<>();
+        for (Message message : messages) {
+            byte[] data = (filterDataSourceType.equals(KEY)) ? message.getLogKey() : message.getLogMessage();
+            String jsonMessage = deserialize(data);
+
+            if (evaluate(jsonMessage)) {
+                filteredMessages.add(message);
+            }
+        }
+        return filteredMessages;
+    }
+
+
+    private boolean evaluate(String jsonMessage) throws FilterException {
+        JsonNode message;
+        try {
+            message = objectMapper.readTree(jsonMessage);
+        } catch (JsonProcessingException e) {
+
```

Comment:
please look at the these empty lines. they should not be there

Reply:
fixed

Reply:
it;s not fixed. i still see the empty line here.

Reply:
fixed